### PR TITLE
podman-remote ps

### PR DESCRIPTION
**Pull request by baude**
add the ability to run ps on containers using the remote client.

Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

File: libpod/container.go [hunk 1]
```diff
@@ -401,6 +401,29 @@ func (t ContainerStatus) String() string {
...
+	case ContainerStateRunning.String():
+		return ContainerStateRunning, nil
+	case ContainerStateStopped.String():
+		return ContainerStateStopped, nil
+	case ContainerStatePaused.String():
+		return ContainerStatePaused, nil
+	case ContainerStateExited.String():
+		return ContainerStateExited, nil
+	default:
+		return ContainerStateUnknown, errors.Errorf("%s not known", status)
```
Comment (jwhonce) [inline libpod/container.go]: Error message should include that this is ContainerState...

Comment (mheon) [reply libpod/container.go]: Also, make this a wrapf - `Wrapf(ErrInvalidArg, "state %q is not known", status)`

File: libpod/container_filter.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	"strconv"
+	"strings"
+
+	"github.com/containers/libpod/pkg/util"
+	"github.com/pkg/errors"
+)
+
+// GenerateContainerFilterFuncs creates functions that can used to filter containers based on set of input
+// filters
+func (r *Runtime) GenerateContainerFilterFuncs(filter, filterValue string) (func(container *Container) bool, error) {
```
Comment (mheon) [inline libpod/container_filter.go]: This shouldn't be in libpod... Adapter would be better

File: cmd/podman/ps.go [hunk 3]
```diff
@@ -2,27 +2,23 @@ package main
 
 import (
 	"fmt"
+	"github.com/containers/libpod/pkg/adapter"
```
Comment (mheon) [inline cmd/podman/ps.go]: Import ordering

**Pull request merged**